addons/sale_loyalty/static/src/views/fields/select_reward_radio/select_reward_radio.js

```diff
@@ -0,0 +1,64 @@
+/** @odoo-module **/
+
+import {registry} from "@web/core/registry";
+import {useService} from "@web/core/utils/hooks";
+import {_lt} from "@web/core/l10n/translation";
+import {standardFieldProps} from "@web/views/fields/standard_field_props";
+import {onWillStart, Component} from "@odoo/owl";
+
+export class SelectRewardRadio extends Component {
+    setup() {
+        this.id = `select_reward_radio_${++SelectRewardRadio.nextId}`;
+        this.orm = useService("orm");
+        onWillStart(async () => {
+            const orderId = this.props.record.data.order_id[0];
+            this.data = await this.orm.call("sale.loyalty.reward.wizard", "get_reward_info", [[orderId]]);
+        });
+    }
+
+    get items() {
+        return this.data || [];
+    }
+
+    get string() {
+        return this.props.record.activeFields[this.props.name].string;
+    }
+
+    get value() {
+        if (this.props.type === "many2one") {
+            return Array.isArray(this.props.value) ? this.props.value[0] : this.props.value;
+        }
+        return null;
+    }
+
+    /**
+     * @param {any} value
+     */
+    onChange(value) {
+        if (this.props.type === "many2one") {
+            this.props.update(value);
+        }
+    }
+}
+
+SelectRewardRadio.additionalClasses = ["o_field_radio"];
+SelectRewardRadio.nextId = 0;
+
+SelectRewardRadio.template = "sale_loyalty.SelectRewardRadioField";
+SelectRewardRadio.props = {
+    ...standardFieldProps,
+    orientation: {type: String, optional: true},
+};
+SelectRewardRadio.defaultProps = {
+    orientation: "vertical",
+};
+
+SelectRewardRadio.displayName = _lt("Radio");
+SelectRewardRadio.supportedTypes = ["many2one"];
+
+SelectRewardRadio.isEmpty = (record, fieldName) => record.data[fieldName] === false;
+SelectRewardRadio.extractProps = ({attrs}) => ({
+    orientation: attrs.options.horizontal ? "horizontal" : "vertical",
+});
```

Comment:
We initially copied the whole code of `RadioField` to make it easier to test but actually it should extend the `RadioField` and only override what is necessary.